pkg/kubectl/resource_printer.go

```diff
@@ -84,15 +86,15 @@ func GetPrinter(format, formatArgument string) (ResourcePrinter, bool, error) {
 // ResourcePrinter is an interface that knows how to print runtime objects.
 type ResourcePrinter interface {
 	// Print receives a runtime object, formats it and prints it to a writer.
-	PrintObj(runtime.Object, io.Writer) error
+	PrintObj(runtime.Object, *watch.EventType, io.Writer) error
 }
 
 // ResourcePrinterFunc is a function that can print objects
-type ResourcePrinterFunc func(runtime.Object, io.Writer) error
+type ResourcePrinterFunc func(runtime.Object, *watch.EventType, io.Writer) error
```

Comment:
As `kubectl -w` needs to somehow support other output types (yaml,json) then using `[][]string` in the interface of the Printer is rather not an option. Internally, HumanReadablePrinter can use whatever it wants (and will use [][]strings/columnDefinition).

If you don't like passing events together with objects to the printer then we will have an extra function setEvent and `kubectl get -w` will do
```go
printer.setEvent(event)
err := printer.PrintObj(obj, writer)
printer.setEvent(nil)
if err != nil ...
```
Is this what you want?